src/peg/mod.rs

```diff
@@ -0,0 +1,210 @@
+//! Convert RVSDGs to PEGs. This is a shortcut to avoid duplicating the work of
+//! analyzing the CFG as loops and ifs, as well as making it easier to do
+//! interoperation between the two dataflow representations.
+//!
+//! # References
+//!
+//! * ["Equality Saturation: A New Approach to Optimization"](https://arxiv.org/abs/1012.1802)
+//! by Tate, Stepp, Tatlock, and Lerner
+
+// todo: remove this once it no longer does anything
+#![allow(dead_code)]
+
+pub(crate) mod simulate;
+
+use crate::rvsdg::{Expr, Id, Operand, RvsdgBody, RvsdgFunction};
+use std::collections::HashMap;
+
+#[cfg(test)]
+mod tests;
+
+/// An expression, expressed using PEGs.
+#[derive(Debug, PartialEq)]
+pub(crate) enum PegBody {
+    /// A pure operation.
+    PureOp(Expr<Id>),
+    /// An argument of the enclosing function.
+    Arg(usize),
+    /// An if statement..
+    Phi(Id, Id, Id),
+    /// A stream that represents a loop.
+    /// The usize is a label (frequently omitted from PEG diagrams).
+    Theta(Id, Id, usize),
+    /// Indexes into a stream.
+    Eval(Id, Id, usize),
+    /// Finds the index of the first value of false in a stream.
+    Pass(Id, usize),
+    /// Does nothing (but useful for creating cycles).
```

Comment:
Update comment to say that it's a layer of indirection for convenience. I guess we could do another pass to remove.